### PR TITLE
Adding unit tests

### PR DESCRIPTION
**Pull request by lucifercr07**
Unit tests being added for the following files:
- mutation-tracker.js
- property-list.js
- request.js
- url.js
- variable.js

### CODE REVIEW

File: test/unit/property-list.test.js [hunk 1]
```diff
@@ -1201,4 +1243,26 @@ describe('PropertyList', function () {
             expect(list.has('key1', 'val1')).to.be.true;
         });
     });
+
+    describe('.toString', function () {
+        it('should handle when unparse method not defined in Type and constructor set to null', function () {
```
[the remark this answers is not in the document]
Comment (lucifercr07) [reply test/unit/property-list.test.js:1271]: We can't test for scenario 2 as there is some bug due to which if the constructor is present `toString` goes into a recursive loop.

File: test/unit/mutation-tracker.test.js [hunk 2]
```diff
@@ -157,6 +157,13 @@ describe('MutationTracker', function () {
 
             expect(tracker.count()).to.eql(0);
         });
+
+        it('should not track invalid track call with no instruction', function () {
```
[the remark this answers is not in the document]
Comment (coditva) [reply test/unit/mutation-tracker.test.js]: A test for valid instruction but invalid payload

**Review by coditva — CHANGES_REQUESTED** [review]

File: test/unit/property-list.test.js [hunk 3]
```diff
@@ -960,6 +960,30 @@ describe('PropertyList', function () {
                 val: 'value2'
             }]);
         });
+
+        it('should throw error when unable to upsert into a list Type that does not support .update()', function () {
```
[the remark this answers is not in the document]
Comment (lucifercr07) [reply test/unit/property-list.test.js:966]: Fixed.

File: test/unit/mutation-tracker.test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (lucifercr07) [reply test/unit/mutation-tracker.test.js]: Fixed.

**Review by coditva — APPROVED** [review]
LGTM! 🎉

File: CHANGELOG.yaml [hunk 4]
```diff
@@ -1,3 +1,9 @@
+master:
+  fixed bugs:
```
Comment (codenirvana) [inline CHANGELOG.yaml:2]: Which bug is fixed here?

File: test/unit/request.test.js [hunk 5]
```diff
@@ -523,6 +523,42 @@ describe('Request', function () {
...
+        it('should remove a header from the request when ignoreCase options set to true', function () {
+            var request = new Request({
+                    header: [
+                        { key: 'foo', value: 'bar' }
+                    ]
+                }),
+                options = {};
+
+            options.ignoreCase = true;
+            request.removeHeader('FOO', options);
```
Comment (codenirvana) [inline test/unit/request.test.js:559]:
```suggestion
            request.removeHeader('FOO', { ignoreCase: true });
```

File: test/unit/request.test.js [hunk 6]
```diff
@@ -581,6 +617,100 @@ describe('Request', function () {
...
+                        {
+                            key: 'some',
+                            value: 'header'
+                        },
+                        {
+                            key: 'other',
+                            value: 'otherheader',
+                            disabled: true
+                        }
+                    ],
```
Comment (codenirvana) [inline test/unit/request.test.js:635]: not required.

File: test/unit/request.test.js [hunk 7]
```diff
@@ -581,6 +617,100 @@ describe('Request', function () {
...
```
Comment (codenirvana) [inline test/unit/request.test.js:667]: not required.

File: test/unit/request.test.js [hunk 8]
```diff
@@ -581,6 +617,100 @@ describe('Request', function () {
...
+                    'type': 'string',
+                    'value': 'postman'
+                }, {
+                    'key': 'password',
+                    'type': 'string',
+                    'value': 'password'
+                }],
+                'type': 'basic'
+            });
+        });
```
Comment (codenirvana) [inline test/unit/request.test.js:713]: Then why's this test here and not in #1101?

**Review by codenirvana — CHANGES_REQUESTED** [review]

File: test/unit/request.test.js (hunk 5 above)
Comment (lucifercr07) [reply test/unit/request.test.js]: Fixed.

File: test/unit/request.test.js (hunk 6 above)
Comment (lucifercr07) [reply test/unit/request.test.js]: Fixed.

File: test/unit/request.test.js (hunk 7 above)
Comment (lucifercr07) [reply test/unit/request.test.js]: Fixed.

File: test/unit/request.test.js (hunk 8 above)
Comment (lucifercr07) [reply test/unit/request.test.js:689]: This check is for `requests.js` method `authorizeUsing` method, https://github.com/postmanlabs/postman-collection/pull/1101 is for `request-auth.js` method `update`.

File: CHANGELOG.yaml (hunk 4 above)
Comment (codenirvana) [reply CHANGELOG.yaml:2]: Which bug is fixed by adding unit tests?

**Review by codenirvana — CHANGES_REQUESTED** [review]

File: CHANGELOG.yaml [hunk 9]
```diff
@@ -1,13 +1,16 @@
 master:
+  chores:
+    - >-
+      GH-1099 Added unit tests for mutation-tracker.js, property-list.js, request.js, url.js
+      and variable.js.
+    - GH-1098 Added unit tests for cookie.js, header.js, request-auth.js, item.js, content-info.js.
```
Comment (codenirvana) [inline CHANGELOG.yaml:6]: Let's follow this order in the changelog:
- new features
- breaking changes
- fixed bugs
- chores

Comment (codenirvana) [reply CHANGELOG.yaml:6]: Also, combine these two and let just have:
`GH-1098, GH-1099 Added unit tests`

**Review by codenirvana — CHANGES_REQUESTED** [review]

File: CHANGELOG.yaml (hunk 9 above)
Comment (lucifercr07) [reply CHANGELOG.yaml]: Done

**Review by codenirvana — APPROVED** [review]

**Pull request merged**